packages/ember-runtime/lib/system/native_array.js

```diff
@@ -138,9 +138,10 @@ var A;
 
 if (Ember.EXTEND_PROTOTYPES === true || Ember.EXTEND_PROTOTYPES.Array) {
   NativeArray.apply(Array.prototype);
-  A = function (arr = []) { return arr; };
+  A = function (arr = []) { return arr || []; };
 } else {
   A = function (arr = []) {
+    if (arr === null) { arr = []; }
```

Comment:
@mixonic Yea that's a good point. I think I was just trying to be more explicit about what was happening. Someone reading the following might feel that it's redundant:

```javascript
A = function (arr = []) {
  if (!arr) { arr = []; }
  return EmberArray.detect(arr) ? arr : NativeArray.apply(arr);
};
```

But it's definitely inconsistent between the two. So let's fix this, which do you think is prefered?